.gitignore

```diff
@@ -1,27 +1,43 @@
+# Please keep workflow- and tool-specific patters out of this
+# file. Just put patterns here that are very generic and we want to
+# avoid adding by mistake and files that are generated by the build
+# system.
+# 
+# Comments from gitignore(1) mention that which file to place a
+# pattern in depends on how the pattern is meant to be used:
+#
+# * Patterns which should be version-controlled and distributed to
+#   other repositories via clone (i.e., files that all developers will
+#   want to ignore) should go into a .gitignore file.
+#
+# * Patterns which are specific to a particular repository but which
+#   do not need to be shared with other related repositories (e.g.,
+#   auxiliary files that live inside the repository but are specific
+#   to one user’s workflow) should go into the $GIT_DIR/info/exclude
+#   file.
+#
+# * Patterns which a user wants Git to ignore in all situations (e.g.,
+#   backup or temporary files generated by the user’s editor of
+#   choice) generally go into a file specified by core.excludesFile in
+#   the user’s ~/.gitconfig. Its default value is
+#   $XDG_CONFIG_HOME/git/ignore. If $XDG_CONFIG_HOME is either not set
+#   or empty, $HOME/.config/git/ignore is used instead.
+#
+
 *~
 **/CMakeFiles/
 **/CMakeCache.txt
 /sql/tests/unit/testoutputs.tmp
 /sql/timescaledb--*.sql
-/data/
-/src/*.o
-/src/*.so
-/src/*.d
-/.vscode/
-/timescaledb.so
+**/*.o
+**/*.so
+**/*.d
 *.bak
 typedef.list
 /test/testcluster
 /test/log
 /test/temp_schedule
-/build
 **/GPATH
 **/GTAGS
 **/GRTAGS
 **/GSYMS
-/.vs
-/compile_commands.json
```

Comment:
`compile_commands.json` is a generated file if you set `-DCMAKE_EXPORT_COMPILE_COMMANDS=true`. We could generalize the pattern for any directory though.